File: src/main/java/net/softbell/bsh/dto/view/advance/TriggerItemCardDto.java
package net.softbell.bsh.dto.view.advance;

import lombok.Getter;
import lombok.Setter;
import net.softbell.bsh.domain.ItemCategoryRule;
import net.softbell.bsh.domain.ItemModeRule;
import net.softbell.bsh.domain.ItemTypeRule;
import net.softbell.bsh.domain.entity.Node;
import net.softbell.bsh.domain.entity.NodeItem;

/**
 * @Author : Bell(bell@softbell.net)
 * @Description : 트리거 등록 및 수정뷰 노드 아이템 카드정보 DTO
 */
@Getter
@Setter
public class TriggerItemCardDto
{
	private Long nodeId;
	private String nodeAlias;
	private Long itemId;
	private String alias;
	private ItemModeRule itemMode;
	private ItemTypeRule itemType;
	private ItemCategoryRule itemCategory;
	
	public TriggerItemCardDto(NodeItem entity)
	{
		// Exception
		if (entity == null)
			return;
		
		// Field
		Node node;
		
		// Init
		node = entity.getNode();
		
		// Convert
		this.nodeId = node.getNodeId();
		this.nodeAlias = node.getAlias();
		this.itemId = entity.getItemId();
		this.alias = entity.getAlias();
		this.itemMode = entity.getItemMode();
		this.itemType = entity.getItemType();
		this.itemCategory = entity.getItemCategory();
	}
}

File: src/main/kotlin/net/softbell/bsh/dto/view/advance/TriggerItemCardDto.kt
package net.softbell.bsh.dto.view.advance

import net.softbell.bsh.domain.ItemCategoryRule
import net.softbell.bsh.domain.ItemModeRule
import net.softbell.bsh.domain.ItemTypeRule
import net.softbell.bsh.domain.entity.NodeItem

/**
 * @author : Bell(bell@softbell.net)
 * @description : 트리거 등록 및 수정뷰 노드 아이템 카드정보 DTO
 */
class TriggerItemCardDto(entity: NodeItem) {
    val nodeId: Long = entity.node.nodeId
    val nodeAlias: String = entity.node.alias
    val itemId: Long = entity.itemId
    val alias: String = entity.alias
    val itemMode: ItemModeRule = entity.itemMode
    val itemType: ItemTypeRule = entity.itemType
    val itemCategory: ItemCategoryRule = entity.itemCategory
}